File: 6kyu/JavaScript/CreatePhoneNumber.js
/*
Write a function that accepts an array of 10 integers (between 0 and 9), that returns 
a string of those numbers in the form of a phone number.

Example
createPhoneNumber([1, 2, 3, 4, 5, 6, 7, 8, 9, 0]) // => returns "(123) 456-7890"
*/

// METHOD 1: use template literals
let createPhoneNumber = nums => {
  if (nums.length === 10) {
    return `(${nums[0]}${nums[1]}${nums[2]}) ${nums[3]}${nums[4]}${nums[5]}-${nums[6]}${nums[7]}${nums[8]}${nums[9]}`
  }
  return 'That is the wrong number of digits for a phone number'
}

// METHOD 2: create a format and use a for loop
let createPhoneNumber = nums => {
  // create format
  let format = '(xxx) xxx-xxxx';
  
  // for loop to update format/mutate the string
  for(let i = 0; i < nums.length; i++) {
    format = format.replace('x', nums[i])
  }
  return format
}

File: 6kyu/Python/CreatePhoneNumber.py
'''
Write a function that accepts an array of 10 integers (between 0 and 9), that returns 
a string of those numbers in the form of a phone number.

Example
createPhoneNumber([1, 2, 3, 4, 5, 6, 7, 8, 9, 0]) // => returns "(123) 456-7890"
'''

# METHOD 1: format strings/template literals
def createPhoneNumber(n):
    return f'({n[0]}{n[1]}{n[2]}) {n[3]}{n[4]}{n[5]}-{n[6]}{n[7]}{n[8]}{n[9]}'

# METHOD 2: use a formatted string, loop through numbers, add them to the format
def createPhoneNumber(n):
    format = '(xxx) xxx-xxxx'
    for num in n:
        # last argument means to replace only the first X
        format = format.replace('x', str(num), 1)
    return format

# METHOD 3: lambda function to do the same thing as method 1
create_phone_number = lambda n : f'({n[0]}{n[1]}{n[2]}) {n[3]}{n[4]}{n[5]}-{n[6]}{n[7]}{n[8]}{n[9]}'
